front/gustavo/tarefa4JS/main.js

```diff
@@ -0,0 +1,127 @@
+        event.querySelector("#check").classList.toggle("hidden")
+        chiken = teste1
+        
+        let foodname =  event.getElementsByClassName('food-title')[0].textContent
+        let foodPrice = event.getElementsByClassName('food-price')[0].textContent.replace("R$ ","").replace(",",".")
+        
+        foodToBy.push(foodname)
+        foodToByP.push(foodPrice)
+
+    }
+    else if(elementId=='drink'){
+        const drinkOptions = document.querySelectorAll('#drink');
+        for(let i=0; i <drinkOptions.length; i++){
+            let childrent_take = drinkOptions[i];
+            if (childrent_take.classList.contains("selected")){
+                childrent_take.classList.remove("selected");
+                childrent_take.querySelector("#check").classList.toggle("hidden");
+                }
+            }
+        let teste2 = event.classList.toggle("selected");
+        event.querySelector("#check").classList.toggle("hidden")
+        drink = teste2
+        let drinkName =  event.getElementsByClassName('food-title')[0].textContent
+        let drinkPrice = event.getElementsByClassName('food-price')[0].textContent.replace("R$ ","").replace(",",".")
+        drinkToBy.push(drinkName)
+        drinkToByP.push(drinkPrice)
+        console.log(drinkPrice)
+        
+    }
+    else{
+        const desertOptions = document.querySelectorAll('#dissert');
+        for(let i=0; i <desertOptions.length; i++){
+            let childrent_take = desertOptions[i];
+            if (childrent_take.classList.contains("selected")){
+                childrent_take.classList.remove("selected");
+                childrent_take.querySelector("#check").classList.toggle("hidden");
+                }
+            }
+        let teste3 = event.classList.toggle("selected");
+        event.querySelector("#check").classList.toggle("hidden")
+        desert = teste3
+        let desertName =  event.getElementsByClassName('food-title')[0].textContent
+        let desertPrice = event.getElementsByClassName('food-price')[0].textContent.replace("R$ ","").replace(",",".")
+
+        desertToBy.push(desertName)
+        desertToByP.push(desertPrice)
+        }
+        let food = (desert === true && drink === true && chiken === true); //verificando se as 3 comidas estão selecionadas
+console.log(drinkToByP)
+verify(food);
+    
+}
+
+function buy(){ //adicionando os textos na tela de cada elemento de acordo com a lista
+
+    telafinal.classList.remove('screenbuy')
+    telafinal.classList.add('fix')
+    let foodFinalScreen = window.document.getElementById('chikenn')
+    let drinksFinalScreen = window.document.getElementById('drinkn')
+    let desertFinalScreen = window.document.getElementById('desertn')
+    foodFinalScreen.textContent = foodToBy[foodToBy.length - 1]
+    drinksFinalScreen.textContent = drinkToBy[drinkToBy.length - 1]
+    desertFinalScreen.textContent = desertToBy[desertToBy.length - 1]
```

Comment:
```suggestion
    const foodFinalScreen = window.document.getElementById('chikenn')
    const drinksFinalScreen = window.document.getElementById('drinkn')
    const desertFinalScreen = window.document.getElementById('desertn')
    foodFinalScreen.textContent = foodToBy[foodToBy.length - 1]
    drinksFinalScreen.textContent = drinkToBy[drinkToBy.length - 1]
    desertFinalScreen.textContent = desertToBy[desertToBy.length - 1]
```

Okay, esse caso aqui é bem especial, como podemos notar, você está sim re-assinalando as variáveis, mas mesmo assim te sugeri usar `const`, por que? Basicamente é porque você não está re-assinalando a variável em si, ou seja, `foodFinalScreen = pipipipopopo`, você está fazendo apenas com um "item" da `foodFinalScreen` (o termo certo seria uma propriedade de `foodFinalScreen`, mas vou chamar de item para facilitar). Logo a `foodFinalScreen` permanece a mesma, ou seja, aponta pro mesmo **objeto**, apenas um de seus "items" foram mudados. Sei que é bastante complexo entender isso, mas qualquer coisa posso explicar melhor depois.